### ISSUE TITLE
Worker Service for ASP.NET Core instruction is not clear

### ISSUE THREAD

**Issue by lettucebo** [aspnet/AspNetCore.Docs#15649]
In the [Package](https://docs.microsoft.com/zh-tw/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-3.0&tabs=visual-studio#package)
says should install `Microsoft.Extensions.Hosting` but in reality I think it should be `Microsoft.AspNetCore.Hosting` for ASP.NET Core can work correctly in Worker Service.

And there is no place to mention about at .NET Core 3 worker service project file `<Project Sdk="Microsoft.NET.Sdk.Worker">` should change to `<Project Sdk="Microsoft.NET.Sdk.Web">` to make asp.net core can be used

Is I'm doing right or I miss something?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c3460b0d-9c60-3a06-ac63-18cf990e53c3
* Version Independent ID: b2dece1f-15b8-03af-6d52-7cc070432b4b
* Content: [Background tasks with hosted services in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-3.0&tabs=visual-studio)
* Content Source: [aspnetcore/fundamentals/host/hosted-services.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/host/hosted-services.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
It's *mostly* ok in its present form if the topic maintains a background tasks (worker) scenario-only topic because ...

* The `Microsoft.Extensions.Hosting` package is the only explicitly-referenced package in a template-based Worker Service app (see the sample app's project file).
* The `Microsoft.Extensions.Hosting` package is what supplies [AddHostedService](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionhostedserviceextensions.addhostedservice) (`Microsoft.Extensions.Hosting.Abstractions` assembly).
* A Worker Service isn't a web app. A web app isn't considered a "background task" for the purposes of describing this scenario. Yes, you can make a web app with a hosted service, but the topic doesn't focus on that scenario.

Here's the problem, which is leftover text from a prior version of the topic that predated the release of the Worker SDK: The *Package* section says that ...

> A package reference to the `Microsoft.Extensions.Hosting` package is added implicitly for ASP.NET Core apps.

... which is misleading/incomplete and incorrect for the Worker SDK. It's true in the sense that the package comes in via the shared framework and a package reference isn't required when using the Web SDK. With the new Worker SDK, that's not the case, as we see in an app created by the Worker Service template (`dotnet new worker`). There's an explicit package reference for it now.

I **_do_** need to make a change to that line. It should call out the situation with that package for the Worker Service scenario with the Worker SDK (i.e., you have to have the package reference).

With regard to a hosted service in a web app that uses the Web SDK, I'll need to check with engineering to see if they want at least a sample of it. That's what we ended up doing over in the Windows Services topic ... we ended up having both a web app-based sample and a purely background task-based sample. The Windows Services topic focuses on the purely background worker approach, but at least devs have a sample app that they can look at to see how to compose a Windows Service with web app capabilities. Here they are if you want to see them ...

https://github.com/aspnet/AspNetCore.Docs/tree/master/aspnetcore/host-and-deploy/windows-service/samples/3.x

I'll get back to this asap. I'm very busy 🏃😅 this week, but I might be able to get to this next week. Thanks for opening this issue.

**Comment by lettucebo** [comment]
I think I have this thought is because my purpose is a `Pure ASP.NET Core web application with some background worker service`

So I follow the first instruction: [Worker Service template](https://docs.microsoft.com/zh-tw/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-3.0&tabs=visual-studio#worker-service-template) in this section says: 
> **The ASP.NET Core Worker Service template** provides a starting point for writing long running service apps. To use the template as a basis for a hosted services app:

So I follow the steps to create the project and can not use ASP.NET Core functionality in the newly create project from Worker Service template.

---

After some research, I just create a ASP.NET Core Web Application and install `[Microsoft.Extensions.Hosting` Package, and follow the doc's instruction to setup, then it worked

**Comment by guardrex** [comment]
Yes ... that's correct.

I think more work is required on this topic similar to what we did over in the Windows Services topic. 

When it came to Windows Services, some devs want a Windows Services app that is purely for background processing (no UI ... no MVC/Razor Pages). Others want a Windows Services app with MVC/Razor Pages. What we did on that topic was to provide **two** samples to cover those scenarios. I think we'll end up doing the same thing with this topic.

I'll should be able to get back to this issue within a few weeks.
